aws/resource_aws_athena_named_query_test.go

```diff
@@ -0,0 +1,91 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/athena"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSAthenaNamedQuery_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAthenaNamedQueryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAthenaNamedQueryConfig(acctest.RandInt(), acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAthenaNamedQueryExists("aws_athena_named_query.foo"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAthenaNamedQueryDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).athenaconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_athena_named_query" {
+			continue
+		}
+
+		input := &athena.GetNamedQueryInput{
+			NamedQueryId: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.GetNamedQuery(input)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case athena.ErrCodeInvalidRequestException:
```

Comment:
The error code here seems quite generic - is this supposed to be an error returned in case the query is gone? Can we make it a bit more specific by any chance? e.g. `isAWSErr(err, athena.ErrCodeInvalidRequestException, "something something")`